cluster-autoscaler/cloudprovider/rancher/rancher_node_group.go

```diff
@@ -0,0 +1,162 @@
+	return int(ng.nodePool().Quantity), nil
+}
+
+// IncreaseSize increases the size of the node group. To delete a node you need
+// to explicitly name it and use DeleteNode. This function should wait until
+// node group size is updated. Implementation required.
+func (ng *rancherNodeGroup) IncreaseSize(delta int) error {
+	return ng.manager.IncreasePoolSize(ng.id, delta)
+}
+
+// DeleteNodes deletes nodes from this node group. Error is returned either on
+// failure or if the given node doesn't belong to this node group. This function
+// should wait until node group size is updated. Implementation required.
+func (ng *rancherNodeGroup) DeleteNodes(nodes []*v1.Node) error {
+	return ng.manager.DeleteNodes(ng.id, nodes)
+}
+
+// DecreaseTargetSize decreases the target size of the node group. This function
+// doesn't permit to delete any existing node and can be used only to reduce the
+// request for new nodes that have not been yet fulfilled. Delta should be negative.
+// It is assumed that cloud provider will not delete the existing nodes when there
+// is an option to just decrease the target. Implementation required.
+func (ng *rancherNodeGroup) DecreaseTargetSize(delta int) error {
+	return fmt.Errorf("sry but I won't decrease the target size ¯\\_(ツ)_/¯")
+}
+
+// Id returns an unique identifier of the node group.
+func (ng *rancherNodeGroup) Id() string {
+	return ng.nodePool().ID
+}
+
+// Debug returns a string containing all information regarding this node group.
+func (ng *rancherNodeGroup) Debug() string {
+	nodePool := ng.nodePool()
+	return fmt.Sprintf("%s - %s (%d:%d)", ng.Id(), nodePool.HostnamePrefix, ng.MinSize(), ng.MaxSize())
+}
+
+// Nodes returns a list of all nodes that belong to this node group.
+func (ng *rancherNodeGroup) Nodes() ([]string, error) {
+	nodes, err := ng.manager.GetCachedNodePoolNodes(ng.id)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeNames := make([]string, len(nodes))
+	for i, node := range nodes {
+		nodeNames[i] = node.ID
+	}
+
+	return nodeNames, nil
+}
+
+// TemplateNodeInfo returns a schedulercache.NodeInfo structure of an empty
+// (as if just started) node. This will be used in scale-up simulations to
+// predict what would a new node look like if a node group was expanded. The returned
+// NodeInfo is expected to have a fully populated Node object, with all of the labels,
+// capacity and allocatable information as well as all pods that are started on
+// the node by default, using manifest (most likely only kube-proxy). Implementation optional.
+func (ng *rancherNodeGroup) TemplateNodeInfo() (*cache.NodeInfo, error) {
+	// TODO we might want to implement this in the future
+	return nil, cloudprovider.ErrNotImplemented
+}
+
+// Exist checks if the node group really exists on the cloud provider side. Allows to tell the
+// theoretical node group from the real one. Implementation required.
+func (ng *rancherNodeGroup) Exist() bool {
+	return true
```

Comment:
Is there a scenario where this can be false? I couldn't find one.

Reply:
In our implementation no, since we're not supporting autoprovisioning of node pools - CA only knows the node pools that were already created at Rancher, so they will always exist.